reedsolomon.go

```diff
@@ -469,21 +469,21 @@ func (r reedSolomon) Split(data []byte) ([][]byte, error) {
 	if len(data) < r.DataShards {
 		return nil, ErrShortData
 	}
+
+	// Calculate number of bytes per shard.
 	perShard := (len(data) + r.DataShards - 1) / r.DataShards
 
-	// Fill data shards.
+	// Pad data to r.Shards*perShard.
+	padding := make([]byte, (r.Shards*perShard)-len(data))
+	data = append(data, padding...)
+
+	// Split into equal-length shards.
 	dst := make([][]byte, r.Shards)
-	for i := 0; i < r.DataShards-1; i++ {
+	for i := range dst {
```

Comment:
this loop could be also be written:
```go
for i := range dst {
    dst[i] = data[i*perShard:(i+1)*perShard]
}
```
This may be more readable, depending on preferred coding style. Up to you.